omod-2.0/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/converter/openmrs2_0/AllergenConverter2_0.java

```diff
@@ -0,0 +1,100 @@
+ * under the License.
+ *
+ * Copyright (C) OpenMRS, LLC.  All Rights Reserved.
+ */
+package org.openmrs.module.webservices.rest.web.v1_0.converter.openmrs2_0;
+
+import org.openmrs.Allergen;
+import org.openmrs.AllergenType;
+import org.openmrs.Concept;
+import org.openmrs.annotation.Handler;
+import org.openmrs.api.context.Context;
+import org.openmrs.module.webservices.rest.SimpleObject;
+import org.openmrs.module.webservices.rest.web.RestConstants;
+import org.openmrs.module.webservices.rest.web.api.RestService;
+import org.openmrs.module.webservices.rest.web.representation.Representation;
+import org.openmrs.module.webservices.rest.web.representation.DefaultRepresentation;
+import org.openmrs.module.webservices.rest.web.resource.api.Converter;
+import org.openmrs.module.webservices.rest.web.response.ConversionException;
+import org.openmrs.module.webservices.rest.web.v1_0.resource.openmrs1_11.ConceptResource1_11;
+import org.apache.commons.lang3.StringUtils;
+import java.util.LinkedHashMap;
+
+/**
+ * An implementation of Converter to be able to create a representation from a User
+ * when User is used in another resource. Currently UserAndPassword doesn't convert User
+ */
+@Handler(supports = Allergen.class, order = 0)
+public class AllergenConverter2_0 implements Converter<Allergen> {
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#getByUniqueId(java.lang.String)
+	 */
+	@Override
+	public Allergen getByUniqueId(String string) {
+		//return Context.getUserService().getUserByUuid(string);
+		return null;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#newInstance(java.lang.String)
+	 */
+	@Override
+	public Allergen newInstance(String type) {
+		return new Allergen();
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#asRepresentation(T, org.openmrs.module.webservices.rest.web.representation.Representation)
+	 */
+	@Override
+	public SimpleObject asRepresentation(Allergen instance, Representation rep) throws ConversionException {
+		SimpleObject allergenObject = new SimpleObject();
+		allergenObject.add("allergenType", instance.getAllergenType());
+		ConceptResource1_11 conceptResource = (ConceptResource1_11) Context.getService(RestService.class).getResourceBySupportedClass(Concept.class);
+		allergenObject.add("codedAllergen", conceptResource.asRepresentation(instance.getCodedAllergen(), rep));
+		allergenObject.add("nonCodedAllergen", instance.getNonCodedAllergen());
+		return allergenObject;
+	}
+	
+	/**
+	 * @see org.openmrs.module.webservices.rest.web.resource.api.Converter#getProperty(T, java.lang.String)
+	 */
+	@Override
+	public Object getProperty(Allergen instance, String propertyName) throws ConversionException {
+		if (propertyName.equals("allergenType")) {
+			return instance.getAllergenType().toString();
+		} else if (propertyName.equals("codedAllergen")) {
+			return instance.getCodedAllergen();
+		} else {
```

Comment:
I think you should explicitly check the nonCodedAllergen property name too because the json might contain other data invalid fields